docs/pipeline_semantics.md

```diff
@@ -0,0 +1,182 @@
+# Pipeline semantics
+
+This document describes the inner workings of a Conduit pipeline, its structure, and behavior. It also describes a
+Conduit message and its lifecycle as it flows through the pipeline.
+
+**NOTE**: Some parts of this document describe behavior that is not yet fully implemented (e.g. DLQs, order of acks).
+For more information see [#383](https://github.com/ConduitIO/conduit/pull/383#pullrequestreview-963196784). This note
```

Comment:
Nitpick: I believe that it was intended to include the link to the PR only, not the PR review.

Reply:
Oko sokolovo 🦅

Reply:
loooool :D